pkg/scalers/gcp_pub_sub_scaler.go

```diff
@@ -19,6 +21,7 @@ import (
 const (
 	defaultTargetSubscriptionSize = 5
 	pubSubStackDriverMetricName   = "pubsub.googleapis.com/subscription/num_undelivered_messages"
+	compositeSubscriptionIDPrefix = "projects/[a-zA-Z0-9-]+/subscriptions/[a-zA-Z0-9-]+"
```

Comment:
```suggestion
	compositeSubscriptionIDPrefix = "projects/[a-z][a-zA-Z0-9-]*[a-zA-Z0-9]/subscriptions/[a-zA-Z][a-zA-Z0-9-_~%+]*"
```

> Project ID can have lowercase letters, digits, or hyphens. It must start with a lowercase letter and end with a letter or number

> [Subscription] ID must start with a letter, and contain only the following characters: letters, numbers, dashes (-), periods (.), underscores (_), tildes (~), percents (%) or plus signs (+). Cannot start with goog.

Reply:
Don't you miss period (.) in the Subscription regex?

Reply:
I've added the period (.) and escaped the '+' and the period sign, just in case.